gemelli/factorization.py

```diff
@@ -276,120 +466,139 @@ def tenals(
+                # zero fibers in the tensor along dim. In practice this
+                # should be rare but can occur in very sparse tensors.
+                denominator[dim][denominator[dim] == 0] = fillna
+                loadings_iter[dim] = loadings_iter[dim] / \
+                    denominator[dim]
+                # If  this dimension is the last in the tensor then
+                # update the eigvals with the new loadings.
+                if dim == len(tensor_dimensions) - 1:
+                    eigvals[component] = norm(loadings_iter[dim])
+                #  update the loadings_iter in dimension (dim)
+                eigvals[component][eigvals[component] == 0] = fillna
+                loadings_iter[dim] = loadings_iter[dim] / \
+                    eigvals[component]
+                loadings[dim][:, component] = loadings_iter[dim]
+            # update the loadings with this iterations loadings
+            for i, loading in enumerate(loadings):
+                loading[:, component] = loadings_iter[i]
+        # MSE of the original tensor and the reconstructed tensor
+        # based on the loadings.
+        mean_squared_error = tensor - mask * \
+            construct_tensor(eigvals, loadings)
+        # Frobenius norm for new reconstructed tensor
+        iteration_tensor_frobenius_norm = norm(mean_squared_error)**2
+        # If the error between this iterations reconstruction and the
+        # intital tensor is below tol_als then  break the iterations.
+        err_conv =  np.sqrt(iteration_tensor_frobenius_norm /
+                            initial_tensor_frobenius_norm)
+        if err_conv < tol_als:
             break
-
-    dist = np.sqrt(normERR / normTE)
     # check that the factorization converged
-    if any(sum(sum(np.isnan(Vn))) > 0 for Vn in V_alt):
+    if any(sum(sum(np.isnan(loading))) > 0 for loading in loadings):
         raise ValueError("The factorization did not converge.",
                          "Please check the input tensor for errors.")
-
-    S_alt = np.diag(S_alt.flatten())
-    # sort the eigenvalues
-    idx = np.argsort(np.diag(S_alt))[::-1]
-    S_alt = S_alt[idx, :][:, idx]
-    # sort loadings
-    loadings = [Vn[:, idx] for Vn in V_alt]
-
-    return loadings, S_alt, dist
-
-
-def RTPM(TE, r, ninit, nitr):
+    # Get the diagonal of the eigvals
+    eigvals = np.diag(eigvals.flatten())
+    # Sort the eigvals and the laodings from
+    # largest to smallest.
+    idx = np.argsort(np.diag(eigvals))[::-1]
+    eigvals = eigvals[idx, :][:, idx]
+    loadings = [loading[:, idx] for loading in loadings]
+
+    return loadings, eigvals, err_conv
+
+
+def robust_tensor_power_method(tensor,
+                               n_components,
+                               n_initializations=50,
+                               max_rtpm_iterations=50,
+                               tol_rtpm=1e-10,
+                               random_state=42):
```

Comment:
42 is an odd default value for `random_state`. Usually these are set to `None` by defualt